storage/url/url.go

```diff
@@ -347,3 +348,12 @@ func parseNonBatch(prefix string, key string) string {
 func hasGlobCharacter(s string) bool {
 	return strings.ContainsAny(s, globCharacters)
 }
+
+func (u *URL) GetCopySource() string {
+	sourceKey := strings.TrimPrefix(u.String(), "s3://")
+	sourceKeyElements := strings.Split(sourceKey, "/")
+	for i, element := range sourceKeyElements {
+		sourceKeyElements[i] = url.QueryEscape(element)
+	}
+	return strings.Join(sourceKeyElements, `/`)
```

Comment:
let's use single-line string representation.

```suggestion
	return strings.Join(sourceKeyElements, "/")
```

Reply:
I updated as your suggestion